### PR TITLE
Use chainId from tx object to created unsignedTx

### PR DESCRIPTION
**Pull request by jiexi**
## Explanation

* Update `getCommonConfiguration` with required `chainId` param
  * Use `chainId` param instead of current selected chainId to generate commonConfiguration object
* Add `prepareUnsignedEthTx` required `chainId` param
* Pass in chainId to `prepareUnsignedEthTx` from relevant tx object
* Pass in chainId to `getCommonConfiguration` from relevant tx object
* Update `approveTransactionsWithSameNonce` param type to include `{chainId: Hex}` to match what extension is actually calling it with

### `@metamask/package-a`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

### `@metamask/package-b`

- **<CATEGORY>**: Your change here
- **<CATEGORY>**: Your change here

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Comment by jiexi** [comment]
going to avoid making any unnecessary changes to TransactionController while we do multichain refactors. Closing this

File: packages/transaction-controller/src/TransactionController.ts [hunk 1]
```diff
@@ -2435,23 +2440,11 @@ export class TransactionController extends BaseControllerV1<
...
-      chain !== RPC &&
-      chain !== NetworkType['linea-goerli'] &&
-      chain !== NetworkType['linea-mainnet']
-    ) {
-      return new Common({ chain, hardfork: HARDFORK });
-    }
-
+  private getCommonConfiguration(chainId: Hex): Common {
     const customChainParams: Partial<ChainConfig> = {
-      name,
```
Comment (adonesky1) [inline packages/transaction-controller/src/TransactionController.ts:2454]: how was this used before 🤔 ?

Comment (jiexi) [reply packages/transaction-controller/src/TransactionController.ts:2454]: I don't believe it is really used

Links below are to a more recent version than the one we use, but still you can see that name isn't really used.

https://github.com/ethereumjs/ethereumjs-monorepo/blob/44e069e41b13eddd8cd09ab62e75f4641eb234d0/packages/tx/src/transactionFactory.ts#L19

https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/common/src/common.ts#L1084

**Comment by jiexi** [comment]
updated to make less changes

**Review by adonesky1 — APPROVED** [review]

**Pull request merged**